### PR TITLE
Migrate reasoner attribute attachment + value predicate ITs to BDD

### PR DESCRIPTION
**Pull request by alexjpwalker**
## What is the goal of this PR?

Where possible, we would like to migrate our reasoner attribute attachment + value predicate tests to BDD.

This PR forms part of the work towards https://github.com/graknlabs/grakn/issues/5721.

## What are the changes implemented in this PR?

This PR will gradually delete test cases from the `test/integration/graql/reasoner` package as they are tested in BDD.

This PR should NOT be merged until verification is updated with the new BDD scenarios, which is the desired outcome of https://github.com/graknlabs/verification/pull/64

### CODE REVIEW

File: dependencies/graknlabs/dependencies.bzl [hunk 1]
```diff
@@ -62,8 +62,8 @@ def graknlabs_simulation():
 def graknlabs_verification():
     git_repository(
         name = "graknlabs_verification",
-        remote = "https://github.com/graknlabs/verification",
-        commit = "787cc1e9c93570ae1397467b13e79ca103721978",  # sync-marker: do not remove this comment, this is used for sync-dependencies by @graknlabs_verification
+        remote = "https://github.com/alexjpwalker/verification",
```
Comment (flyingsilverfin) [inline dependencies/graknlabs/dependencies.bzl:65]: don't forget to revert before merge :)

File: test/behaviour/resolution/ResolutionSteps.java [hunk 2]
```diff
@@ -79,7 +80,14 @@ public void for_graql_query(String graqlQuery) {
 
     @Then("in reasoned keyspace, answer size is: {number}")
     public void reasoned_keyspace_answer_size_is(final int expectedCount) {
-        resolution.manuallyValidateAnswerSize(queryToTest, expectedCount);
+        final Transaction testTx = reasonedSession.transaction(Transaction.Type.READ);
```
Comment (flyingsilverfin) [inline test/behaviour/resolution/ResolutionSteps.java:83]: we should use consistent terminology now: `test` is not in use anymore - we use `reasoned` everywhere instead

File: test/integration/graql/reasoner/reasoning/ValuePredicateIT.java [hunk 3]
```diff
@@ -481,93 +218,21 @@ public void derivingResources_requireNotHavingSpecificValue() {
...
-                    .forEach(p -> assertNotEquals(p.first(), p.second()));
-            List<ConceptMap> negationAnswers = tx.execute(Graql.parse(negationVersion).asGet());
-            assertCollectionsNonTriviallyEqual(answers, negationAnswers);
         }
     }
 
+    // TODO: The "restricting inferred concepts simultaneously by type and value" bit is migrated;
+    // the "filtering inferred concepts by an abstract type does nothing" part should be migrated somewhere else.
     @Test
     public void derivingResources_requireAnEntityToHaveTwoDistinctResourcesOfNotAbstractType() {
```
Comment (flyingsilverfin) [inline test/integration/graql/reasoner/reasoning/ValuePredicateIT.java:237]: delete

Comment (alexjpwalker) [reply test/integration/graql/reasoner/reasoning/ValuePredicateIT.java:237]: done!

File: test/behaviour/resolution/ResolutionSteps.java (hunk 2 above)
Comment (alexjpwalker) [reply test/behaviour/resolution/ResolutionSteps.java:83]: Consider it consistencised.

**Pull request merged**